DiscordChatExporter.Core/Discord/Data/ThreadChannel.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Linq;
+using System.Text.Json;
+using DiscordChatExporter.Core.Discord.Data.Common;
+using DiscordChatExporter.Core.Utils.Extensions;
+using JsonExtensions.Reading;
+
+namespace DiscordChatExporter.Core.Discord.Data;
+
+// https://discord.com/developers/docs/resources/channel#channel-object-example-thread-channel
+public partial record ThreadChannel(
```

Comment:
I wonder if it would be easier to just use the existing `Channel` class for threads too. It has some extra stuff, but it should be nullable anyway. Might have to redesign the `ChannelCategory` concept though, since now it's more of a parent, which can either be a category (for channels) or a channel (for threads). Don't have to do it now, just thinking out loud.

Reply:
Yeah, `ChannelCategory` is an interesting one. My thinking around how threads would be displayed in the GUI is pretty similar to categories - channels with threads get a dropdown for all of their threads, and threads can be selected/deselected as with channels.

There are a few issues with that approach, though:

- When selecting a channel for export, does it also select all threads belonging to it? If so, does the inverse apply when deselecting the channel?
- The same problem could occur when selecting individual/multiple threads. My thinking, though, is that thread selection should not affect the export status of the parent channel in any way.
- Because a channel would also be a dropdown, what sort of UI adjustment would need to happen in order to support selection/deselection of said parent channel in a non-confusing way? Currently, you just click the channel to mark it for export -- I suppose this could still work if we made the dropdown arrow behave like a small button?

Just some thoughts for the future, although perhaps these would be better suited for #999.